zipline-loader/src/nativeTest/kotlin/app/cash/zipline/loader/URLSessionZiplineHttpClientTest.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2022 Block, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.zipline.loader
+
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.assertTrue
+import kotlinx.coroutines.runBlocking
+import okio.IOException
+import platform.Foundation.NSURLSession
+
+/**
+ * This tests our Kotlin/Native URLSessionZiplineHttpClient. Unfortunately this test is not enabled
+ * by default as we don't have an equivalent to MockWebServer for Kotlin/Native.
```

Comment:
ktor server + arraydeque of simple response sealed class will probably get us very far here

Reply:
ooooh yeah that’s a great idea. 
https://github.com/cashapp/zipline/issues/715